storage/replica.go

```diff
@@ -3376,6 +3377,59 @@ func (r *Replica) maybeGossipSystemConfig() {
 	r.systemDBHash = hash
 }
 
+// maybeGossipNodeLiveness gossips information for all node liveness
+// records stored on this range. To scan and gossip, this replica
+// must hold the lease to a range which contains some or all of the
+// node liveness records. After scanning the records, it checks
+// against what's already in gossip and only gossips records which
+// are out of date.
```

Comment:
Actually I had to back out this change. Too many tests rely on the existing behavior that we re-gossip identical data. This just isn't worth figuring out for this PR.